### PR TITLE
Changed Event System

### PR DESCRIPTION
**Pull request by Hammer86gn**
The event system now uses Reflection and uses fewer lambdas to be more organized

### CODE REVIEW

**Comment by Hammer86gn** [comment]
In order to use this have a class that implements IEventListener. Then make a public method with the only parameter the event you want to listen to. Then Annotate the method you made with EventWatcher. Finally register the class in the EventRegister. Please note you can have as many listeners in a class as you like

**Comment by MarcusSlover** [comment]
Coolio.

**Comment by Reasonlesss** [comment]
Looks good

**Comment by Reasonlesss** [comment]
Looks good

File: src/main/java/io/github/codeutilities/event/EventRegister.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class EventRegister {
+    private static EventRegister instance;
+
+    private Map<String, List<Method>> listeners = new HashMap<>();
```
Comment (tk2217) [inline src/main/java/io/github/codeutilities/event/EventRegister.java:15]: This could probably be a `Map<Class, List<Method>>`

**Review by tk2217 — COMMENTED** [review]
Code wise I feel that this is well done although I do have some issues with the currently exposed paradigm. Personally I feel that at least in some cases the old system was a bit better, especially with just adding single listeners. Even with these criticisms, having a way to register listeners through annotated methods may allow for cleaner code in many places. My recommendation is either to continue with the old system or use some sort of hybrid between the two.

*Note: I'm not going to approve this or request changes due to wanting feedback and having large architectural comments.*

**Comment by tk2217** [comment]
iirc @Reasonlesss had some other stuff she dm'd me

**Comment by Hammer86gn** [comment]
> iirc @Reasonlesss had some other stuff she dm'd me

what would those be?

**Pull request closed without merging**